### ISSUE TITLE
Investigate test_bip32_speed test

### ISSUE THREAD

**Issue by tsusanka** [trezor/trezor-firmware#663]
Currently, test_public_ckd and test_private_ckd fail on Trezor T on a real device. They pass on emulator. We should:

- investigate why they fail on TT
- if they fail due to wrong thresholds, modify them
- test against T1
- make sure test_cache test is skipped on TT rightfully

**Comment by tsusanka** [comment]
T1 times for public ckd:

```
DEPTH 0 EXPECTED DELAY 0.26 REAL DELAY 0.04191470146179199
DEPTH 1 EXPECTED DELAY 0.52 REAL DELAY 0.07903242111206055
DEPTH 2 EXPECTED DELAY 0.78 REAL DELAY 0.07900428771972656
DEPTH 3 EXPECTED DELAY 1.04 REAL DELAY 0.07890653610229492
DEPTH 4 EXPECTED DELAY 1.3 REAL DELAY 0.07896304130554199
DEPTH 5 EXPECTED DELAY 1.56 REAL DELAY 0.07896828651428223
DEPTH 6 EXPECTED DELAY 1.82 REAL DELAY 0.07896995544433594
DEPTH 7 EXPECTED DELAY 2.08 REAL DELAY 0.0789649486541748
```

TT times for public ckd:

```
DEPTH 0 EXPECTED DELAY 0.26 REAL DELAY 1.040006399154663
DEPTH 1 EXPECTED DELAY 0.52 REAL DELAY 1.092951774597168
DEPTH 2 EXPECTED DELAY 0.78 REAL DELAY 1.1309714317321777
DEPTH 3 EXPECTED DELAY 1.04 REAL DELAY 1.16996169090271
DEPTH 4 EXPECTED DELAY 1.3 REAL DELAY 1.2089407444000244
DEPTH 5 EXPECTED DELAY 1.56 REAL DELAY 1.2490229606628418
DEPTH 6 EXPECTED DELAY 1.82 REAL DELAY 1.2999458312988281
DEPTH 7 EXPECTED DELAY 2.08 REAL DELAY 1.3399829864501953
```

**Comment by tsusanka** [comment]
Hm, a fast PoC does correspond to that idea - the time is more or less the same. Will investigate further (mainly if the root_index is found properly).

**Comment by tsusanka** [comment]
Result of test_bip32_speed.py's test_cache:

```
NOCACHE TIME 15.447206258773804
CACHED TIME 11.01799201965332
NOCACHE TIME 11.576215744018555
CACHED TIME 11.04405164718628
NOCACHE TIME 11.562138795852661
CACHED TIME 11.054014205932617
NOCACHE TIME 11.560360193252563
CACHED TIME 11.039029121398926
NOCACHE TIME 11.587116956710815
CACHED TIME 11.017045259475708
```

**Comment by tsusanka** [comment]
I have created an issue to tackle this https://github.com/trezor/trezor-firmware/issues/671.

I have skipped the tests for now for TT in https://github.com/trezor/trezor-firmware/commit/c6d6d6c1d7a04bacc7b50538425823a6313f738b, since it does not make sense to have a test that fails and we ignore it. We will put it back after we fix #671.
